wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -877,6 +880,15 @@ private WorkspaceImpl doCreateWorkspace(WorkspaceConfig config,
                                                      .setAttributes(attributes)
                                                      .setTemporary(isTemporary)
                                                      .build();
+
+        // Register default SSH keypair for this workspace.
+        try {
+            this.sshManager.generatePair(EnvironmentContext.getCurrent().getSubject().getUserId(), "workspace", workspace.getId());
```

Comment:
- about events yes why not but then in api-ssh machine-ext-server is not at the right "level" as it needs to be at workspace level
it would mean that api-ssh depends on api-workspace (in my PR it's the opposite)

- on removal of ssh key yes make sense